### PR TITLE
Add 'An overview of the Odin programming language and its features.'

### PR DESCRIPTION
**Pull request by beastmatser**
## What does this PR do?
Add resource

### Description

The official language overview from the [Odin website](https://odin-lang.org)

### Why is this valuable (or not)?

It adds a book for a programming language which is not yet referenced in this repo.

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md)
- [x] Search for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction)

### CODE REVIEW

File: books/free-programming-books-langs.md [hunk 1]
```diff
@@ -1654,6 +1655,11 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [Octave Programming](https://en.wikibooks.org/wiki/Octave_Programming_Tutorial) - Wikibooks
 
 
+### Odin
+
+* [The Odin language overview](https://odin-lang.org/docs/overview/)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1660]: Don't invent titles! Use the title given by the resource. In this case `Overview | Odin Programming Language`. (check other titles to see how to properly handle the pipe character in markdown.)

Comment (beastmatser) [reply books/free-programming-books-langs.md]: At https://odin-lang.org/docs/ it is described as 'An overview of the Odin programming language and its features.', would this be valid as a name?

Comment (eshellman) [reply books/free-programming-books-langs.md]: That's a description, not a title. I took the title from the html title element

Comment (beastmatser) [reply books/free-programming-books-langs.md]: I changed it to what you recommended, thank you for taking the time to review my submission. :)

Comment (eshellman) [reply books/free-programming-books-langs.md]: markdown wants you to use `\|`

Comment (beastmatser) [reply books/free-programming-books-langs.md]: I just noticed it as well, I already made the change

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**